addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -237,30 +237,32 @@ function align(editor, mode) {
  * @param {string} mode 'color' or 'backgroundColor'
  */
 function colorElement(element, color, mode) {
-    const newClassName = element.className
+    let newClassName = element.className
         .replace(mode === 'color' ? TEXT_CLASSES_REGEX : BG_CLASSES_REGEX, '')
         .replace(/\btext-gradient\b/g, '') // cannot be combined with setting a background
         .replace(/\s+/, ' ');
-    element.className !== newClassName && (element.className = newClassName);
     element.style['background-image'] = '';
     if (mode === 'backgroundColor') {
         element.style['background'] = '';
     }
     if (color.startsWith('text') || color.startsWith('bg-')) {
         element.style[mode] = '';
-        element.classList.add(color);
+        newClassName += (newClassName ? " " : "") + color;
     } else if (isColorGradient(color)) {
         element.style[mode] = '';
         if (mode === 'color') {
             element.style['background'] = '';
             element.style['background-image'] = color;
-            element.classList.add('text-gradient');
+            newClassName += (newClassName ? " " : "") + "text-gradient";
         } else {
             element.style['background-image'] = color;
         }
     } else {
         element.style[mode] = color;
     }
+    // Replacing the classes only once at the end to have only one class
+    // mutation in the current history step.
+    element.className = newClassName;
```

Comment:
I don't think we should fix the issue like that (although I am very tempted to remove the awful line above). We should not have to think about the mutations an edition code will trigger.

I am not sure what the solution is though, I don't understand your paragraph in the previous commit:
> - When we stop hovering (`color_leave` event), we revert the current
> step and so the mutation records are filtered first. For classes, there
> will be filtered this way (the element currently has the `text-gradient`
> class):
>   - 1st mutation: the oldValue is in the element classes and vice versa
>    -> no `excludedClasses`
>       => the record is discarded (`continue`)
>   - 2nd mutation: the current element class is not in oldValue
>     -> `text-gradient` is added in `excludedClasses` and it is not in
>    `renderingClasses` so the condition is false
>       => the record is kept

What does "we revert the current step and so the mutation records are filtered first. " actually means? We don't call colorElement back? We relying on the previously recorded mutations? If that is so, why are filtering them again? (Maybe I don't understand anything about the way it works here, you tell me ^^).

Also I would squash both commits. It makes sense for a commit to mention its previous one (and I think here you did as RDE suggested in case of a "revert" but RDE answered generically, without knowing what this was actually about I think) but in this case:
- Your revert reintroduces a bug (at least I think?) ->  we should not reintroduce bugs, either fix them the right way first and then revert the other problematic commit which created other bugs, or:
- In this case, I would squash because I cannot understand this commit without the previous one (I always had to jump back and forth between your previous commit message and this diff here to try and understand).